### ISSUE TITLE
'window is not defined' - accessing CMIS from server side

### ISSUE THREAD

**Issue by wimco** [agea/CmisJS#32]
Is it possible to use CmisJS without having a browser open?

I'm trying to use it in a headless situation from a Meteor Node.js server, but get the following:

```
ReferenceError: window is not defined
at Object.a.createSession (app/server/3rdparty/cmis-0.2.0.js:5:20407)
```

**Comment by jaawerth** [comment]
I haven't used Meteor, but I seem to recall that it does some bizarre custom stuff with modules. CmisJS current goes into "node" mode [by checking if CommonJS is in use and window is undefined](https://github.com/agea/CmisJS/blob/master/lib/cmis.js#L25-L26), which could be what's causing the issue. Also, did you install via npm?

**Comment by wimco** [comment]
No, presently I just call it directly as a js script.  It works fine in the browser, and it looks to work from server side when I comment out lines 1124-1132 of the CmisJS script:

```
    // http://visionmedia.github.io/superagent
    var request;

    // if running on node.js require superagent
    if (typeof module !== 'undefined' && module.exports) {
      request = require('superagent');
    } else {
      request = window.superagent;
    }
```
And then explicitly create a global variable "<b>request</b>" to instantiate superagent.  Since there's no window available to satisfy the if statement on line 1128.

Will do some more tests with this case and  post back whether it works right.

**Comment by jaawerth** [comment]
If you look online, there seem to be a few guides out there for integrating npm packages into Meteor - if you want to be able to get updates as you develop, that's personally what I'd look into first, but YMMV.

**Comment by wimco** [comment]
I agree that that would be the better way.

I tried that at first months ago, ran into issues with connecting to Alfresco 4.2, and the NPM module for CmisJS was 2 versions behind the one on Git.  So I thought it would help to use the latest version.  It didn't, as the problem was that Alfresco 4.2 did not fully support the CMIS 1.1 spec, and when I changed to Alfresco 5, everything worked.

So it's perhaps time to refactor my code to use the NPM again...
Thanks
